lumen/ai/llm.py

```diff
@@ -67,14 +61,8 @@ async def invoke(
 
         output = None
         for r in range(self.retry):
-            try:
-                output = await client(messages=messages, **kwargs)
-                break
-            except Exception as e:
-                print(f"Error encountered: {e}")
-                if 'response_model' in kwargs:
-                    kwargs['response_model'] = response_model
-                messages = messages + [{"role": "system", "content": f"You just encountered the following error, make sure you don't repeat it: {e}" }]
+            output = await self.run_client(model_key, messages, **kwargs)
+            break
```

Comment:
Don't understand this, where is the error handling?

Reply:
I implemented in a separate PR:
https://github.com/holoviz/lumen/pull/543/files#diff-d56253257f94b517eb742b1213dec9b7ccb3fafa7416be4044987ccd54ed3c83L91

https://github.com/holoviz/lumen/pull/543/files#diff-d56253257f94b517eb742b1213dec9b7ccb3fafa7416be4044987ccd54ed3c83R69-R79

Reply:
I'm not convinced it is, the exception handler will only catch it when it bubbles all the way up to whatever triggered the execution, so it will never retry right?!

Reply:
I suppose you're right; it won't retry that specific part.

I added it back.